scripts/update_and_rebuild_libmesh.sh

```diff
@@ -5,24 +5,50 @@ DIAGNOSTIC_LOG="libmesh_diagnostic.log"
 # Set go_fast flag if "--fast" is found in command line args.
 for i in "$@"
 do
+  shift
   if [ "$i" == "--fast" ]; then
     go_fast=1;
-    break;
+  fi
+
+  if [[ "$i" == "-h" || "$i" == "--help" ]]; then
+    help=1;
+  fi
+
+  if [ "$i" == "--skip-submodule-update" ]; then
+    skip_sub_update=1;
+  else # Remove the skip submodule update argument before passing to libMesh configure
+    set -- "$@" "$i"
```

Comment:
I have not seen this `set --` thing before. Isn't `shift` used for removing arguments from `$@`?

Reply:
I pieced this together from an Internet search. This just adds the item back to the front of the list since shift takes it off.